cpp/src/parquet/arrow/arrow_reader_writer_test.cc

```diff
@@ -649,6 +649,28 @@ TYPED_TEST(TestParquetIO, SingleColumnRequiredWrite) {
   ASSERT_NO_FATAL_FAILURE(this->ReadAndCheckSingleColumnFile(*values));
 }
 
+TYPED_TEST(TestParquetIO, EmptyColumnTable) {
+  std::shared_ptr<Array> values;
+  ASSERT_OK(NonNullArray<TypeParam>(0, &values));
+  std::shared_ptr<Table> table = MakeSimpleTable(values, false);
+
+  this->ResetSink();
+  ASSERT_OK_NO_THROW(WriteTable(*table, ::arrow::default_memory_pool(), this->sink_,
+                                values->length(), default_writer_properties()));
+
+  std::shared_ptr<Table> out;
+  std::unique_ptr<FileReader> reader;
+  ASSERT_NO_FATAL_FAILURE(this->ReaderFromSink(&reader));
+  ASSERT_NO_FATAL_FAILURE(this->ReadTableFromFile(std::move(reader), &out));
+  ASSERT_EQ(1, out->num_columns());
+  ASSERT_EQ(0, out->num_rows());
+
+  std::shared_ptr<ChunkedArray> chunked_array = out->column(0);
+  ASSERT_EQ(1, chunked_array->num_chunks());
```

Comment:
If there is one chunk, then does it actually exercise the bug?

Reply:
I think you're right

Reply:
https://github.com/apache/arrow/pull/5393/commits/985e553b5f32a57346e987c6b152f50d3c441ff7#diff-8fbac668446292c63462df42a5c96737R667

Odd: even when the test writes a table with zero chunks, the table which is read has a single (empty) chunk

Reply:
I see. It doesn't need to block this PR.